### DISCUSSION TITLE
Welcome to CleanMinimalApi Discussions!

### DISCUSSION THREAD

**Discussion by stphnwlsh** [stphnwlsh/CleanMinimalApi#4] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by natandovale** [comment]
Hey!! this project is really good!! I based it on a project, but I have a question, the discards that you use in some places, are they used for what purpose?
![image](https://user-images.githubusercontent.com/66981502/202713770-60653067-9720-418c-9f1d-888ac29caa20.png)

**Reply by stphnwlsh** [reply]
Hey!  Thanks heaps, glad you like it.  Will eventually update it to .NET 7 as well, when time permits.

As for the discards, it's mostly for style rather than substance.  I like using them to remind me that the action actually returns a value, but I just have no reason to make it a variable.  I started it via an editor config, but now I just like it.

You can drop it from your own project but you'd have to edit the `.editorconfig` file.  It's line 179.

```
# C# Unnecessary code rules
[*.{cs,csx,cake}]
csharp_style_unused_value_expression_statement_preference = unused_local_variable:suggestion
dotnet_diagnostic.IDE0058.severity = suggestion
csharp_style_unused_value_assignment_preference = unused_local_variable:suggestion
dotnet_diagnostic.IDE0059.severity = suggestion
```

**Reply by stphnwlsh** [reply]
I should say their best purpose is is discarding a forced out parameter. Something like this, if you don't need the outputted date, more the knowledge that it can be parsed.

```
DateTime.TryParse(dateString, out _)
```

**Reply by natandovale** [reply]
ahh I understood, thanks for answering!!
